src/Nethermind/Nethermind.Facade/Eth/BlockForRpc.cs

```diff
@@ -77,7 +74,6 @@ public BlockForRpc(Block block, bool includeFullTransactionData, ISpecProvider s
         Size = _blockDecoder.GetLength(block, RlpBehaviors.None);
         StateRoot = block.StateRoot;
         Timestamp = block.Timestamp;
-        TotalDifficulty = block.TotalDifficulty ?? 0;
```

Comment:
this is still usefull for pre-merge networks.

Maybe `TotalDifficulty = block.Difficulty.IsZero ? null : block.TotalDifficulty ?? 0;` and make `TotalDifficulty` nullable and ignore nulls?

Reply:
It was on main Ethereum.